### PR TITLE
Collection tab w/ Favorites folder #1346

### PR DESCRIPTION
**Pull request by Kamran-Khan-Gene**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: wallet/lib/pages/detailed_asset_view/owner_view_view_model.dart [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import 'dart:async';
 import 'dart:ui';
+
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/detailed_asset_view/owner_view_view_model.dart:3]: Why extra line?

File: wallet/lib/pages/detailed_asset_view/owner_view_view_model.dart [hunk 2]
```diff
@@ -28,14 +31,16 @@ class OwnerViewViewModel extends ChangeNotifier {
   final AudioPlayerHelper audioPlayerHelper;
   final VideoPlayerHelper videoPlayerHelper;
   final ShareHelper shareHelper;
+  final FavoritesChangeNotifier favoritesChangeNotifier;
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/detailed_asset_view/owner_view_view_model.dart:34]: Instead here . Provide it from the MaterialApp.

File: wallet/lib/pages/detailed_asset_view/widgets/pdf_placeholder.dart [hunk 3]
```diff
@@ -9,32 +9,41 @@ class PdfPlaceHolder extends StatelessWidget {
...
 
   const PdfPlaceHolder({
     Key? key,
     required this.nftUrl,
     required this.nftName,
     required this.thumbnailUrl,
+    this.showIcon = true,
   }) : super(key: key);
 
   Widget getPdfThumbnailFromUrl() {
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/detailed_asset_view/widgets/pdf_placeholder.dart:22]: Cannot we make one widget for all Thmbnails?

File: wallet/lib/pages/detailed_asset_view/widgets/video_placeholder.dart [hunk 4]
```diff
@@ -12,34 +12,43 @@ class VideoPlaceHolder extends StatelessWidget {
...
 
   const VideoPlaceHolder({
     Key? key,
     required this.nftUrl,
     required this.nftName,
     required this.thumbnailUrl,
+    this.showIcon = true,
   }) : super(key: key);
 
   Widget getVideoThumbnailFromUrl() {
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/detailed_asset_view/widgets/video_placeholder.dart:25]: Looks same like PDF thumnail.

File: wallet/lib/pages/purchase_item/purchase_item_view_model.dart [hunk 5]
```diff
@@ -35,8 +37,11 @@ class PurchaseItemViewModel extends ChangeNotifier {
     required this.repository,
     required this.shareHelper,
     required this.accountPublicInfo,
+    required this.favoritesChangeNotifier,
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/purchase_item/purchase_item_view_model.dart:40]: Provide it from the top. Then there will be only one notifier.

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]
Also no test for this code?

File: wallet/lib/pages/detailed_asset_view/owner_view_view_model.dart (hunk 1 above)
Comment (Kamran-Khan-Gene) [reply wallet/lib/pages/detailed_asset_view/owner_view_view_model.dart:3]: dart formatter is adding this extra line.

File: wallet/lib/pages/detailed_asset_view/widgets/pdf_placeholder.dart (hunk 3 above)
Comment (Kamran-Khan-Gene) [reply wallet/lib/pages/detailed_asset_view/widgets/pdf_placeholder.dart:22]: image


image


image

Comment (Kamran-Khan-Gene) [reply wallet/lib/pages/detailed_asset_view/widgets/pdf_placeholder.dart:22]: they are pretty different,,

File: wallet/lib/pages/detailed_asset_view/owner_view_view_model.dart [hunk 6]
```diff
@@ -263,10 +264,21 @@ class OwnerViewViewModel extends ChangeNotifier {
...
     likedByMe = !likedByMe;
     isLiking = false;
+    final favoritesChangeNotifier = GetIt.I.get<FavoritesChangeNotifier>();
     if (temp && likesCount > 0) {
       likesCount = likesCount - 1;
+      repository.deleteNFTFromFavorites(recipeId);
+      favoritesChangeNotifier.removeFromFavorites(recipeId: recipeId);
     } else {
       likesCount = likesCount + 1;
+      final favoriteModel = FavoritesModel(
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/detailed_asset_view/owner_view_view_model.dart:274]: make a separate method. A function should do only one functionality.

File: wallet/lib/pages/settings/widgets/delete_dialog.dart [hunk 7]
```diff
@@ -22,98 +22,95 @@ class DeleteDialog {
...
+                  ),
+                  Container(
+                    margin: EdgeInsets.symmetric(horizontal: 20.w),
+                    height: 40.h,
+                    child: Row(
+                      crossAxisAlignment: CrossAxisAlignment.stretch,
+                      children: [
+                        Expanded(
+                          child: buildButton(
+                            title: "yes",
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/settings/widgets/delete_dialog.dart:71]: Change it to `Localekey.yes.tr()`

**Comment by Kamran-Khan-Gene** [comment]
@kjawadDeveloper2 kindly review my PR.

**Comment by Kamran-Khan-Gene** [comment]
@kjawadDeveloper2 kindly review my PR.

File: wallet/lib/pages/detailed_asset_view/owner_view_view_model.dart [hunk 8]
```diff
@@ -265,11 +266,28 @@ class OwnerViewViewModel extends ChangeNotifier {
...
 
+  void addToFavAndUpdateView({required String cookBookID, required String recipeId}) {
+    final favoriteModel = FavoritesModel(
+      id: recipeId,
+      cookbookId: cookBookID,
+      type: NftType.TYPE_RECIPE.name,
+      dateTime: DateTime.now().millisecondsSinceEpoch,
+    );
+    repository.insertNFTInFavorites(favoriteModel);
+    GetIt.I.get<FavoritesChangeNotifier>().addToFavorites(favoritesModel: favoriteModel);
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/detailed_asset_view/owner_view_view_model.dart:284]: Not good approach. Add it as a dependency.

File: wallet/lib/pages/detailed_asset_view/owner_view_view_model.dart [hunk 9]
```diff
@@ -265,11 +266,28 @@ class OwnerViewViewModel extends ChangeNotifier {
...
+  }
+
+  void removeFromFavAndUpdateView({required String recipeId}) {
+    GetIt.I.get<FavoritesChangeNotifier>().removeFromFavorites(recipeId: recipeId);
```
Comment (kjawadDeveloper2) [inline wallet/lib/pages/detailed_asset_view/owner_view_view_model.dart:288]: Not good approach. Add it as a dependency.

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**